lib/src/rules/diagnostic_describe_all_properties.dart

```diff
@@ -0,0 +1,248 @@
+          if (isPrivate(v.name)) {
+            continue;
+          }
+          if (v.declaredElement.isStatic) {
+            continue;
+          }
+          if (isWidgetProperty(v.declaredElement.type)) {
+            continue;
+          }
+          if (_isOverridingMember(v.declaredElement)) {
+            continue;
+          }
+
+          properties.add(v.name);
+        }
+      }
+    }
+
+    if (properties.isEmpty) {
+      return;
+    }
+
+    // todo (pq): move up to top when we're not counting anymore.
+    var debugFillProperties = node.getMethod('debugFillProperties');
+    if (debugFillProperties == null) {
+      ++classesWithPropertiesButNoDebugFill;
+    }
+
+    var debugDescribeChildren = node.getMethod('debugDescribeChildren');
+    if (debugFillProperties == null && debugDescribeChildren == null) {
+      return;
+    }
+
+    // Remove any defined in debugFillProperties.
+    removeReferences(debugFillProperties, properties);
+
+    // Remove any defined in debugDescribeChildren.
+    removeReferences(debugDescribeChildren, properties);
+
+    // Flag the rest.
+    properties.forEach(rule.reportLint);
+
+//    // todo (pq): remove before landing
+//    for (var prop in properties) {
+//      var line = lineInfo.getLocation(prop.offset).lineNumber;
+//      var prefix =
+//          'https://github.com/flutter/flutter/blob/master/packages/flutter/';
+//      var path = node.element.source.fullName.split('packages/flutter/')[1];
+//      print('| [$path:$line]($prefix$path#L$line) | ${node.name}.$prop |');
+//      ++debugPropertyCount;
+//    }
+  }
+
+  void removeReferences(
+      MethodDeclaration method, List<SimpleIdentifier> properties) {
+    if (method == null) {
+      return;
+    }
+    DartTypeUtilities.traverseNodesInDFS(method.body)
+        .whereType<SimpleIdentifier>()
+        .forEach((p) {
+      var debugName;
+      var name;
+      if (p.name.startsWith('debug') && p.name.length > 5) {
+        debugName = p.name;
+        name = '${p.name[5].toLowerCase()}${p.name.substring(6)}';
+      } else {
+        name = p.name;
+        debugName = 'debug${p.name[0].toUpperCase()}${p.name.substring(1)}';
+      }
+      properties.removeWhere(
+          (property) => property.name == debugName || property.name == name);
+    });
+  }
+
+  var collectionInterfaces = <InterfaceTypeDefinition>[
+    new InterfaceTypeDefinition('List', 'dart.core'),
+    new InterfaceTypeDefinition('Map', 'dart.core'),
+    new InterfaceTypeDefinition('LinkedHashMap', 'dart.collection'),
+    new InterfaceTypeDefinition('Set', 'dart.core'),
+    new InterfaceTypeDefinition('LinkedHashSet', 'dart.collection'),
+  ];
+
+  bool isWidgetProperty(DartType type) {
+    if (DartTypeUtilities.implementsInterface(type, 'Widget', '')) {
+      return true;
+    }
+    if (type is ParameterizedType &&
+        DartTypeUtilities.implementsAnyInterface(type, collectionInterfaces)) {
+      // todo (pq): improve....
```

Comment:
Make todo more specific about how to improve.
Should we improve based on is we do see false positives in the flutter code base with more complex collection types?